packages/react/emotion-utils/src/Stack.tsx

```diff
@@ -1,50 +1,52 @@
 /** @tossdocs-ignore */
 /** @jsxImportSource @emotion/react */
-import { forwardRef, ReactElement, Ref } from 'react';
-
+import { ComponentPropsWithRef, forwardRef, ReactElement } from 'react';
 import { Flex, FlexOptions } from './flex';
 import { gutter, GutterOptions } from './gutter';
-import { AsProps, AxisDirection, InferenceHTMLElement } from './types';
+import { AsProps, StringElementType } from './types';
+
+type StackOptions = Pick<FlexOptions, 'align' | 'justify'> &
+  Partial<Pick<GutterOptions, 'direction' | 'selector'>> & {
+    gutter?: number;
+  };
 
-interface StackProps<T extends keyof JSX.IntrinsicElements = 'div'>
-  extends AsProps<T>,
-    Omit<FlexOptions, 'direction'>,
-    Omit<GutterOptions, 'direction' | 'space'> {
-  direction?: AxisDirection;
-  gutter?: number;
-}
+type StackProps<T extends StringElementType = StringElementType> = AsProps<T> & StackOptions;
 
-type StackReturnType = <T extends keyof JSX.IntrinsicElements = 'div'>(
```

Comment:
I renamed `StackReturnType` to `BaseStackType` like `BaseFlexType` because this is not Return type